go-controller/pkg/ovn/routeimport/route_import.go

```diff
@@ -343,7 +323,27 @@ func (c *controller) syncNetwork(network string) error {
 	if info == nil {
 		return nil
 	}
-	router := info.GetNetworkScopedGWRouterName(c.node)
+
+	// get the table from the network VRF. Note we go to netlink for this as
+	// source of truth instead of using c.tables cache which is just a hint for
+	// syncRouteUpdate. This avoids implementing a more complicated logic to
+	// mantain c.tables
+	table, err := c.getRoutingTableForNetwork(network)
+	if err != nil {
+		return fmt.Errorf("failed to get VRF table from network: %w", err)
+	}
+	if table == noTable {
+		return nil
+	}
+
+	// sneakily set the hint for syncRouteUpdate. Handles this sequence of events:
```

Comment:
I don't like that we have to do it but I guess this is still better than figuring out the network based on the tableID from a route update..

Reply:
I think this is a good compromise